ExtCore/Control.Cps.Result.fs

```diff
@@ -0,0 +1,265 @@
+    member __.Delay f
+        : ProtectedResultStateContFunc<'State, 'T, 'Error, 'K> =
+        f ()
+
+    // M<'T> * ('T -> M<'U>) -> M<'U>
+    member inline __.Bind (m : ProtectedResultStateContFunc<_,_,_,_>, k : 'T -> ProtectedResultStateContFunc<_,_,_,_>) =
+        fun state cont ->
+            m state <| fun result ->
+                match result with
+                | Error error ->
+                    Error error
+                    |> cont
+                | Ok (result, state) ->
+                    k result state cont
+
+    // M<'T> -> M<'T> -> M<'T>
+    // or
+    // M<unit> -> M<'T> -> M<'T>
+    member inline __.Combine (r1 : ProtectedResultStateContFunc<_,_,_,_>, r2 : ProtectedResultStateContFunc<_,_,_,_>)
+        : ProtectedResultStateContFunc<'State, 'T, 'Error, 'K>=
+        fun state cont ->
+            r1 state <| fun result ->
+                match result with
+                | Error error ->
+                    Error error
+                    |> cont
+                | Ok ((), state) ->
+                    r2 state cont
+
+    // M<'T> * (exn -> M<'T>) -> M<'T>
+    member inline __.TryWith (body : ProtectedResultStateContFunc<_,_,_,_>, handler : exn -> ProtectedResultStateContFunc<_,_,_,_>)
+        : ProtectedResultStateContFunc<'State, 'T, 'Error, 'K> =
+        fun state cont ->
+            try body state cont
+            with ex ->
+                handler ex state cont
+
+    // M<'T> -> M<'T> -> M<'T>
+    member inline __.TryFinally (body : ProtectedResultStateContFunc<_,_,_,_>, handler)
+        : ProtectedResultStateContFunc<'State, 'T, 'Error, 'K> =
+        fun state cont ->
+            try body state cont
+            finally
+                handler ()
+
+    // 'T * ('T -> M<'U>) -> M<'U> when 'U :> IDisposable
+    member this.Using (resource : ('T :> System.IDisposable), body : 'T -> ProtectedResultStateContFunc<_,_,_,_>)
+        : ProtectedResultStateContFunc<'State, 'U, 'Error, 'K> =
+        this.TryFinally (body resource, (fun () ->
+            if not <| isNull (box resource) then
+                resource.Dispose ()))
+
+    // (unit -> bool) * M<'T> -> M<'T>
+    member this.While (guard, body : ProtectedResultStateContFunc<_,_,_,_>)
+        : ProtectedResultStateContFunc<'State, unit, 'Error, 'K> =
+        if guard () then
+            this.Bind (body, (fun () -> this.While (guard, body)))
+        else
+            this.Zero ()
+
+    // seq<'T> * ('T -> M<'U>) -> M<'U>
+    // or
+    // seq<'T> * ('T -> M<'U>) -> seq<M<'U>>
+    member this.For (sequence : seq<_>, body : 'T -> ProtectedResultStateContFunc<_,_,_,_>)
+        : ProtectedResultStateContFunc<'State, unit, 'Error, 'K> =
+        this.Using (sequence.GetEnumerator (), fun enum ->
+            this.While (
+                enum.MoveNext,
+                this.Delay (fun () ->
+                    body enum.Current)))
+
+
+/// <summary>
+/// </summary>
+[<AutoOpen>]
+module ResultWorkflowBuilders =
+
+    //
```

Comment:
I don't see any value in all these empty comments. Maybe remove all of them everywhere?